src/main/java/frc/robot/subsystems/Tale/CoralMech.java

```diff
@@ -0,0 +1,239 @@
+import com.ctre.phoenix6.StatusSignal;
+import com.ctre.phoenix6.configs.CurrentLimitsConfigs;
+import com.ctre.phoenix6.configs.Slot0Configs;
+import com.ctre.phoenix6.configs.SoftwareLimitSwitchConfigs;
+import com.ctre.phoenix6.configs.TalonFXSConfiguration;
+import com.ctre.phoenix6.controls.DutyCycleOut;
+import com.ctre.phoenix6.controls.PositionVoltage;
+import com.ctre.phoenix6.controls.VelocityVoltage;
+import com.ctre.phoenix6.hardware.TalonFXS;
+import com.ctre.phoenix6.signals.MotorArrangementValue;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+import edu.wpi.first.epilogue.Logged;
+import edu.wpi.first.math.controller.ArmFeedforward;
+import edu.wpi.first.units.measure.*;
+import edu.wpi.first.wpilibj.simulation.SingleJointedArmSim;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+/** Arm subsystem using TalonFXS with Minion motor */
+@Logged(name = "CoralMechSubsystem")
+public class CoralMech extends SubsystemBase {
+  // Constants
+  private final int canID = 18;
+  private final double gearRatio = 4; // Gear ratio
+  private final Voltage kP = Voltage.ofBaseUnits(0, Volts);
+  private final Voltage kI = Voltage.ofBaseUnits(0, Volts);
+  private final Voltage kD = Voltage.ofBaseUnits(0, Volts);
+  private final AngularVelocity maxVelocity = RadiansPerSecond.of(1); // rad/s
+  private final AngularAcceleration maxAcceleration = RadiansPerSecondPerSecond.of(1); // rad/s²
+  private final boolean brakeMode = true;
+  private final Angle forwardSoftLimit = Angle.ofBaseUnits(0, Radians); // max angle in radians
+  private final Angle reverseSoftLimit = Angle.ofBaseUnits(0, Radians); // min angle in radians
+  private final boolean enableStatorLimit = true;
+  private final Current statorCurrentLimit = Current.ofBaseUnits(100, Amps);
+  private final boolean enableSupplyLimit = true;
+  private final Current supplyCurrentLimit = Current.ofBaseUnits(60, Amps);
+  private final Distance armLength = Distance.ofBaseUnits(1, Meters); // meters
+
+  // Feedforward
+  private final ArmFeedforward feedforward =
+      new ArmFeedforward(
+          0, // kS
+          0, // kG
+          0, // kV
+          0 // kA
+          );
+
+  // Motor controller
+  private final TalonFXS motor;
+  private final PositionVoltage positionRequest;
+  private final VelocityVoltage velocityRequest;
+  private final StatusSignal<Angle> positionSignal;
+  private final StatusSignal<AngularVelocity> velocitySignal;
+  private final StatusSignal<Voltage> voltageSignal;
+  private final StatusSignal<Current> statorCurrentSignal;
+  private final StatusSignal<Temperature> temperatureSignal;
+
+  // voltage
+  private DutyCycleOut m_coralPercentOutput = new DutyCycleOut(0);
+
+  // Simulation
+  //  private final SingleJointedArmSim armSim;
+
+  /** Creates a new Arm Subsystem. */
+  public CoralMech() {
+    // Initialize motor controller
+    motor = new TalonFXS(canID, "canivore");
```

Comment:
It would be nice to have canbusname as a variable that is set to "canivore" near where canID gets set.